processor/transformprocessor/internal/logs/logs.go

```diff
@@ -109,9 +109,21 @@ func newPathGetSetter(path []common.Field) (common.GetSetter, error) {
 	case "flags":
 		return accessFlags(), nil
 	case "trace_id":
-		return accessTraceID(), nil
+		if len(path) == 1 {
+		    return accessTraceID(false), nil
+		}
+		if path[1].Name == "string" {
+		    return accessTraceID(true), nil
+		}
+		return accessTraceID(false), nil
 	case "span_id":
-		return accessSpanID(), nil
+		if len(path) == 1 {
+		    return accessSpanID(false), nil
+		}
+		if path[1].Name == "string" {
+		    return accessSpanID(true), nil
+		}
+		return accessSpanID(false), nil
```

Comment:
Please update these cases to use a switch statement like "resource".  Also, please create new accessor functions.  The new accessor functions should return strings in the Getter and accept strings in the Setter.  [This PR contains the old string code for Trace and Span IDs and can be used again.](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10903).

Reply:
Correct you'll need to add it back, same with ParseSpanID.  Both functions are in the PR I linked in `parser.go`